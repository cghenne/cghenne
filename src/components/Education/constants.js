export const education = [
  {
    graduation_year: 2008,
    school: 'Paul Lambin Institute',
    location: 'Brussels, Belgium',
    description: `BSc Computer Science (web services, infographic
      and artificial intelligence)`
  },
  {
    graduation_year: 2005,
    school: 'High School "Lyc√©e de Berlaymont"',
    location: 'Waterloo, Belgium',
    description: `Sciences and mathematics`
  }
];

export type SchoolT = {
  graduation_year: number,
  school: string,
  location: string,
  description: string
};
