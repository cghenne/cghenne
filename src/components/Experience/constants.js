export const jobs = [
  {
    start_date: 'January 2018',
    end_date: null,
    position: 'Senior Full-Stack Engineer',
    company: 'LifeWorks',
    location: 'Toronto, CA',
    description: `LifeWorks transfered me to their Toronto office to direct the
      development support in the Eastern Time Zone, continue the evolution and
      innovation of the product and create opportunities to build out a North
      America-based development team.`,
    tech_stack: null
  },
  {
    start_date: 'July 2016',
    end_date: 'December 2017',
    position: 'Lead Front-End Engineer',
    company: 'LifeWorks',
    location: 'London, UK',
    description: `In addition to front-end development,
      I performed hiring, training, and mentoring
      for the front-end team. I was also involved in project
      development and oversight and have helped to improve the
      operating platform.`,
    tech_stack: null
  },
  {
    start_date: 'July 2015',
    end_date: 'July 2016',
    position: 'Front-End Engineer',
    company: 'LifeWorks',
    location: 'London, UK',
    description: `LifeWorks: A Global Employee Engagement
      Platform serving 50,000 companies with over 15 million users.`,
    tech_stack: `JavaScript, React, AngularJS, HTML, CSS, Sass,
      Redux, Flow, Node.js, Webpack, Gulp, NPM, Clojure, ClojureScript,
      Jenkins, Vagrant`
  },
  {
    start_date: 'May 2012',
    end_date: 'July 2015',
    position: 'Senior Web Developer',
    company: 'Vanilla Active',
    location: 'London, UK',
    description: `Vanilla Active is a digital agency offering services to
      agencies and brands. We designed and implemented promotional web
      sites directly with brands and worked closely with advertising
      and marketing agencies.`,
    tech_stack: 'HTML, CSS, JavaScript, AngularJS, ColdFusion, PHP, Wordpress',
    print_margin: true
  },
  {
    start_date: 'November 2011',
    end_date: 'March 2012',
    position: 'Excel Expert',
    company: 'Wild Media Ltd',
    location: 'London, UK',
    description: `Wild Media Ltd is a startup in the new
      media area. As an Excel expert and
      database manager, I was in charge of optimising
      and automating the logistics department.`,
    tech_stack: null
  },
  {
    start_date: 'July 2011',
    end_date: 'November 2011',
    position: 'Consultant',
    company: 'De Valck Consultants by Altran',
    location: 'Brussels, Belgium',
    description: `De Valck Consultants is part of Altran group,
      the European leader in innovation consulting.
      It is the dedicated brand to Quality Assurance
      for the Life Science industry. I worked as an
      IS Validation Engineer at Baxter Healthcare.`,
    tech_stack: null
  },
  {
    start_date: 'December 2008',
    end_date: 'June 2011',
    position: 'Web Developer',
    company: 'Jules Bordet Institute',
    location: 'Brussels, Belgium',
    description: `The BrEAST data centre is a specialised clinical
      trials unit created in order to conduct large,
      international Phase III studies in breast cancer
      aiming to register new drugs. I worked as a web
      and application developer, in IT support and
      maintenance, and in validation of computer systems
      developed in-house (GCP, 21 CRF part 11).`,
    tech_stack: 'HTML, CSS, JavaScript, PHP, Java, Oracle SQL, MySQL'
  }
];

export type JobT = {
  start_date: string,
  end_date: ?string,
  position: string,
  company: string,
  location: string,
  description: ?string,
  tech_stack: ?string,
  print_margin?: boolean
};
