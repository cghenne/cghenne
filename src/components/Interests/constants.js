export const interests = [
  {
    type: 'sports',
    label: 'Sports',
    description: 'Outdoor sports, squash, badminton'
  },
  {
    type: 'Music',
    label: 'Music',
    description: 'Piano, Guitar'
  },
  {
    type: 'other',
    label: 'Other',
    description: 'Travelling'
  }
];

export type InterestT = {
  type: string,
  label: string,
  description: string
};
