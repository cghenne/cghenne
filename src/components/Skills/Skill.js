// @flow
import React from 'react';
import './Skills.css';

type Props = {
  label: string,
  list: Array<string>
};

const Skill = ({ label, list }: Props) => {
  return (
    <div className="skill">
      <h4>{label}</h4>
      <ul>
        {list.map((item: string, index: number) => <li key={index}>{item}</li>)}
      </ul>
    </div>
  );
};

Skill.displayName = 'Skill';

export default Skill;
