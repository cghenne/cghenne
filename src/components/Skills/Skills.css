html,
body,
div,
span,
applet,
object,
iframe,
h1,
h2,
h3,
h4,
h5,
h6,
p,
blockquote,
pre,
a,
abbr,
acronym,
address,
big,
cite,
code,
del,
dfn,
em,
img,
ins,
kbd,
q,
s,
samp,
small,
strike,
strong,
sub,
sup,
tt,
var,
b,
u,
i,
center,
dl,
dt,
dd,
ol,
ul,
li,
fieldset,
form,
label,
legend,
table,
caption,
tbody,
tfoot,
thead,
tr,
th,
td,
article,
aside,
canvas,
details,
embed,
figure,
figcaption,
footer,
header,
hgroup,
menu,
nav,
output,
ruby,
section,
summary,
time,
mark,
audio,
video {
  margin: 0;
  padding: 0;
  border: 0;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
}

article,
aside,
details,
figcaption,
figure,
footer,
header,
hgroup,
menu,
nav,
section {
  display: block;
}

body {
  line-height: 1;
}

ol,
ul {
  list-style: none;
}

blockquote,
q {
  quotes: none;
}

blockquote:before,
blockquote:after,
q:before,
q:after {
  content: '';
  content: none;
}

table {
  border-collapse: collapse;
  border-spacing: 0;
}

.skills {
  grid-area: skills;
}

.skill {
  display: grid;
  grid-template-columns: repeat(1, 1fr);
  grid-template-rows: auto;
  grid-template-areas: 'title' 'list';
  margin-bottom: 20px;
}
.skill:last-child {
  margin-bottom: 0;
}
@media (min-width: 61.25em) {
  .skill {
    grid-template-columns: repeat(12, 1fr);
    grid-template-areas: 'title title title . list list list list list list list list';
  }
}
.skill h4 {
  grid-area: title;
  font-weight: 700;
  text-transform: uppercase;
  font-size: 14px;
  color: #58595a;
  margin-bottom: 10px;
}
@media (min-width: 61.25em) {
  .skill h4 {
    margin-bottom: 0;
  }
}
.skill ul {
  grid-area: list;
}
