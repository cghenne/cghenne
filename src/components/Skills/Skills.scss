@import 'styles/utils.scss';

.skills {
  grid-area: skills;
}

.skill {
  display: grid;
  grid-template-columns: repeat(1, 1fr);
  grid-template-rows: auto;
  grid-template-areas:
    "title"
    "list";

  margin-bottom: $spacing;

  &:last-child {
    margin-bottom: 0;
  }

  @include mq($from: desktop) {
    grid-template-columns: repeat(12, 1fr);
    grid-template-areas:
      "title title title . list list list list list list list list";
  }

  @media print {
    grid-template-columns: repeat(12, 1fr);
    grid-template-areas:
      "title title title . list list list list list list list list";
  }

  h4 {
    grid-area: title;
    font-weight: $font-weight-bold;
    text-transform: uppercase;
    font-size: $font-size-small;
    color: $color-secondary;
    margin-bottom: $spacing / 2;

    @include mq($from: desktop) {
      margin-bottom: 0;
    }

    @media print {
      font-size: $font-size-x-small;
      font-weight: $font-weight-regular;
    }
  }

  ul {
    grid-area: list;
  }
}
