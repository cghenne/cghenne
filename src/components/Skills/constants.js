export const skills = [
  {
    type: 'languages',
    label: 'Languages',
    list: ['JavaScript ES6/ES7', 'CSS3/Sass', 'HTML5']
  },
  {
    type: 'frameworks-libraries',
    label: 'Frameworks & Libraries',
    list: [
      'React',
      'Redux',
      'Jest',
      'Flow',
      'Node.js',
      'Express',
      'AngularJS 1.x'
    ]
  },
  {
    type: 'tools',
    label: 'Tools',
    list: ['Webpack', 'NPM', 'Gulp', 'Travis CI']
  },
  {
    type: 'databases',
    label: 'Databases',
    list: ['MongoDB', 'MySQL']
  },
  {
    type: 'other',
    label: 'Other',
    list: [
      'Git',
      'Jira',
      'Agile workflows',
      'Single Page Application',
      'Responsive Web Design',
      'Mobile First Design',
      'Cross-Browser Compatibility',
      'Accessiblity'
    ]
  }
];

export type SkillT = {
  type: string,
  label: string,
  list: Array<string>
};
