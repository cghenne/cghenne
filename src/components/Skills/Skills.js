// @flow
import React from 'react';
import Skill from './Skill';
import { skills } from './constants';
import type { SkillT } from './constants';
import './Skills.css';

const Skills = () => {
  return (
    <div className="box skills">
      <h3>Skills</h3>
      {skills.map((skill: SkillT) => <Skill {...skill} key={skill.type} />)}
    </div>
  );
};

Skills.displayName = 'Skills';

export default Skills;
