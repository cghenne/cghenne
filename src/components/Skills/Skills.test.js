import React from 'react';
import { shallow } from 'enzyme';
import Skills from './Skills';
import { skills } from './constants';

describe('Skills', () => {
  const wrapper = shallow(<Skills />);

  it('should render without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render title', () => {
    expect(wrapper.find('h3').text()).toEqual('Skills');
  });

  it('should render the box with a class of box and skills', () => {
    expect(wrapper.find('div').prop('className')).toEqual('box skills');
  });

  it('should render a Skill component for each skill in the list of skills', () => {
    expect(wrapper.find('Skill').length).toEqual(skills.length);
  });
});
