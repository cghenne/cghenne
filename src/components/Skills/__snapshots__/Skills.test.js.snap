// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skills should render without crashing 1`] = `
<div
  className="box skills"
>
  <h3>
    Skills
  </h3>
  <Skill
    key="languages"
    label="Languages"
    list={
      Array [
        "JavaScript ES6/ES7",
        "CSS/SASS",
        "HTML",
      ]
    }
    type="languages"
  />
  <Skill
    key="frameworks-libraries"
    label="Frameworks / Libraries"
    list={
      Array [
        "React",
        "Redux",
        "React Native",
        "Node.js",
        "Express",
        "Sequelize",
        "Jest",
        "Flow",
      ]
    }
    type="frameworks-libraries"
  />
  <Skill
    key="tools"
    label="Tools"
    list={
      Array [
        "Webpack",
        "NPM/Yarn",
        "Travis CI",
        "CircleCI",
      ]
    }
    type="tools"
  />
  <Skill
    key="databases"
    label="Databases"
    list={
      Array [
        "PostgreSQL",
        "MongoDB",
        "MySQL",
      ]
    }
    type="databases"
  />
  <Skill
    key="other"
    label="Other"
    list={
      Array [
        "Git",
        "Jira",
        "Agile workflows",
        "AWS",
      ]
    }
    type="other"
  />
</div>
`;
